### ISSUE TITLE
release the saga monitor store + selectors as a separate package

### ISSUE THREAD

**Issue by yelouafi** [yelouafi/redux-saga-devtool#4]
Releasing the non UI separately would benefit to to other redux devtools. the monitor's selectors could present normalized data into multiple forms (effects by parent/child relation, by action/reaction, task fork/join relation, channels inter-communication etc)

**Comment by zalmoxisus** [comment]
According to the README it will endup as a browser extension and could be integrated with Redux DevTools extension as well. As the extension is in a separate process, we'll have to stringify the data from the client side everytime to get it there. So there're 2 possible implementations:
1. To add a subscribe callback to saga monitor store, and whenever it is changed, send the state to the extension. To implement that, we'd have to remove `connect` from the current containers and pass the state from the root container as a prop we can change.
2. Pass a custom `dispatch` callback to `createSagaMonitor`, which will get the actions and will just send them to the extension, and dispatch them there.

**Comment by yelouafi** [comment]
The saga monitor uses its own internal store (not the app store). The monitor is connected to the saga middleware and receives monitor events directly from it (not via redux actions). 

But there maybe still an issue because monitor events may contain non serializable data (mainly functions & symbols)

**Comment by zalmoxisus** [comment]
@yelouafi, yes, because the store is independent we are able to implement it on the extension side and just pass actions from the client side. We can serialize functions, symbols and other special types [as Redux Extension does](https://github.com/zalmoxisus/redux-devtools-extension/releases/tag/v2.8.4), just it will be slower.

**Comment by zalmoxisus** [comment]
Basically, we'll pass a custom dispatch to `createSagaMonitor` from client side like:

``` js
const dispatch = (action) => { window.postMessage(action, '*') }
```

So, `effectTriggered`, `effectResolved`... will just call this function instead of `store.dispatch`.

Then on the extension side we're getting the action and do `store.dispatch(action)`.

**Comment by zalmoxisus** [comment]
The problem is that parsed objects would be different [here](https://github.com/yelouafi/redux-saga-devtools/blob/master/src/store/selectors.js#L120). I guess we should somehow add action id [here](https://github.com/yelouafi/redux-saga-devtools/blob/master/src/store/createSagaMonitor.js#L46-L51) and compare ids. This would also allow not to include the action payload in the effects multiple times, which could be huge for serialization.

**Comment by yelouafi** [comment]
the action object is stored from [here](https://github.com/yelouafi/redux-saga-devtools/blob/master/src/store/createSagaMonitor.js#L88). We also compare by ref [here](https://github.com/yelouafi/redux-saga-devtools/blob/master/src/store/selectors.js#L72) to find sagas who took an action.

If a unique id should be assigned to actions, I think it should be done on the redux-saga side because actually the saga monitor (residing in the extension process) will see 2 different objects (one in `actionDispatched` and another in `effectResolved` for `take` effects).

Another possible way (perhaps) is for the saga monitor to use the fact that `effectResolved` for `take` effects will occur immediately after an `actionDispatched` events. So when we get an `effectResolved` for `take`s we could assume it relates to the lasted dispatched action and save the info directly to the store (instead of computing it in demand)

**Comment by zalmoxisus** [comment]
> Another possible way (perhaps) is for the saga monitor to use the fact that effectResolved for take effects will occur immediately after an actionDispatched events...

If that would work, it would be awesome. We could add a parameter with a custom `createActionId` which would return the action ID from Redux DevTools lifted store. So, we could select effects for actions right from Redux Monitor.
